core/src/cats/cats.h

```diff
@@ -1004,7 +1004,8 @@ class BareosDb : public BareosDbQueryEnum {
 
 /* Use for better error location printing */
 #define UPDATE_DB(jcr, cmd) UpdateDB(__FILE__, __LINE__, jcr, cmd, 1)
-#define UPDATE_DB_NO_AFR(jcr, cmd) UpdateDB(__FILE__, __LINE__, jcr, cmd, 0)
+#define UPDATE_DB_NO_AFFECTED_ROWS(jcr, cmd) \
+  UpdateDB(__FILE__, __LINE__, jcr, cmd, 0)
```

Comment:
While this is a lot better than before, the naming is still bad.

Just from the naming I'd guess that `UPDATE_DB()` runs an update query on the database without any further checks (besides making sure the query did not fail).
For `UPDATE_DB_NO_AFFECTED_ROWS()` I'd guess it does the same as `UPDATE_DB()` but won't return the number of affected rows.

I think it would be a lot easier to understand if we changed `UPDATE_DB()` and `BareosDB::UpdateDB()` to
* not to take the additional parameter for minimum affected rows
* return the number of affected rows as an `int` and `-1` on error.

This way the existing calls to `UPDATE_DB()` that look like `if(UPDATE_DB(...))` would have to be changed to `if(UPDATE_DB(...) > 0)` (i.e. at least one row was affected) and `UPDATE_DB_NO_AFFECTED_ROWS()` could be done with either `>= 0` or `!= -1`.

Same is probably true for `INSERT_DB()`, which currently fails if you do a multi-row insert like `INSERT INTO x (col_name) VALUES (row1_value), (row2_value), (row3_value);` or `INSERT INTO y SELECT * FROM z;`. So for keeping the existing behaviour you'd need `if(INSERT_DB(...) == 1)`.